### DISCUSSION TITLE
❌ Améliorer le fichier Excel d'import massif

### DISCUSSION THREAD

**Discussion by ghost** [betagouv/ma-cantine#1041] [Idées]
Deux améliorations suggérées par une utilisatrice (I. L.) le 10/02 :
- avoir un menu déroulant proposant les différentes valeurs acceptées dans les colonnes acceptant un nombre réduit de valeurs (colonnes Secteur, Mode de production, Mode de gestion et Secteur économique). Cela évite de courir le risque que l'utilisateur ne lise pas les explications données au sujet du tableur sur la page "Créer des cantines via CSV" et importe ensuite un fichier qui ne soit pas accepté par la base de données, ce qui s'était passé. Ex. : comme elle n'avait pas lu les explications, elle avait renseigné "Santé" dans la colonne Secteur au lieu de mettre "Médical".

- mettre des valeurs en français plutôt qu'en anglais pour les colonnes Mode de gestion et Secteur économique

**Comment by ghost** [comment]
@alemangui & @hfroot, que pensez-vous de ces suggestions?

**Reply by hfroot** [reply]
Je pense qu'on a développé cet outil pour un type d'utilisateur qui est confortable en lisant la documentation sur la page, et si c'est pas le cas c'est mieux pour l'utilisateur d'utiliser le route normal pour créer des cantines, qui a déjà les listes déroulantes et tout bien écrit en français. 
Ou sinon on doit réfléchir les personnes cibles pour l'outil.

**Reply by ghost** [reply]
Est-ce incompatible de laisser la documentation et de faire aussi des menus déroulants dans le fichier Excel? Si ça ne l'est pas, qu'est-ce qui te fait penser que ce n'est pas pertinent?

**Reply by hfroot** [reply]
Il y a quelques problèmes reliés avec les menus déroulants :
- on doit MAJ deux fichiers : la version excel et libre office
- si les listes change, on doit se souvenir de MAJ les listes dans ces deux fichiers
- dans quelques champs, c'est possible de mettre plusieurs valeurs. Excel ne soutien pas ça (au point que je connais)

Il me semble que il y ait une question plus générale à poser : pourquoi est-ce que les utilisateurs préfère utiliser le fichier excel et pas notre UI ? Est-ce qu'on doit/peut améliorer le UI pour le rendre plus agréable à utiliser ?

L'outil de l'import massif est pour des gens qui ont déjà les données au format numérique, alors ils ne doivent pas remplir le fichier excel à la main.

**Reply by alemangui** [reply]
My two cents : On n'a jamais vraiment écrit noir sur blanc les cas d'usage et les types d'utilisateur cible de cette fonctionnalité. Personnellement j'avais en tête des gestionnaires ayant déjà un progiciel de suivi qui permet de faire des exports de données, et qui peuvent donc configurer ces exports pour produire des fichiers lisibles pour nous. Ceci veut dire que l'utilisateur ne touche pas directement au CSV, ne le rédige pas iel même. 

Par contre je serai curieux de savoir si on gagnerait à améliorer/fluidifier l'entrée des données conventionnelle pour que les utilisateurs ayant une poignée de cantines et n'ayant pas une connaissance technique ou un progiciel se sentent à l'aise en rentrant leurs infos via le site.

**Reply by ghost** [reply]
Merci pour vos précisions, @hfroot et @alemangui. En particulier, je ne savais pas que vous aviez envisagé que l'utilisateur ne rédige pas lui-même le fichier CSV. C'est désormais plus clair pour moi.

Bilan :
=> Je ne proposerai l'outil d'import massif qu'aux responsables d'achats de SRC ou aux responsables de département/région qui ont : 1. plus d'une 10aine de cantines à créer, 2. utilisent un progiciel qui permet de faire des exports de données en CSV.
=> J'inviterai désormais les gestionnaires de cuisines centrales, qui souvent ne sont pas très à l'aise avec Excel et parfois même avec l'informatique en général, à créer plutôt individuellement leurs cantines.

Question : si vous aviez imaginé que l'utilisateur ne rédige pas lui-même le fichier CSV, est-ce que des valeurs différentes de celles que vous avez indiquées peuvent être acceptées? Ex. : si sur le logiciel de l'utilisateur il est indiqué "Santé" à la place de "Médical" ou "administratif" à la place de "Administration", est-ce que ces valeurs, identiques pour le sens mais différentes sur la forme, peuvent être acceptées par notre base de données?

**Reply by ghost** [reply]
=> Suggestion à abandonner car inutile : le tableau d'import massif ne sera pas (ou que très rarement) rempli à la main.
